### PR TITLE
feat: edit and delete urls

### PR DESCRIPTION
**Pull request by PraiseTy**

### CODE REVIEW

File: .eslintrc.json
Comment (PraiseTy) [inline .eslintrc.json:1]: Commenting here because I don't think I'll push tonight. This rule was annoying me so I turned it off

File: src/routes/url.ts
Comment (PraiseTy) [inline src/routes/url.ts:1]: will resolve this comment tomorrow

File: src/controllers/url.ts [hunk 1]
```diff
@@ -71,4 +71,70 @@ const getUrl = async (req: Request, res: Response) => {
...
+      return res.status(HTTP_ERRORS.BAD_REQUEST).json({ error: 'Url cannot be found by Id' });
+    }
+
+    return res.status(HTTP_ERRORS.OK).json({
+      message: 'Url updated successfully',
+      data: {
+        id,
+        customName: customName || updateUrls.customName,
+        originalUrl: url || updateUrls.originalUrl,
+        shortUrl: shortUrls || updateUrls.shortUrl,
```
Comment (greatertomi) [inline src/controllers/url.ts]:
```suggestion
        customName: updateUrls.customName,
        originalUrl: updateUrls.originalUrl,
        shortUrl: updateUrls.shortUrl,
```

File: src/controllers/url.ts [hunk 2]
```diff
@@ -71,4 +71,70 @@ const getUrl = async (req: Request, res: Response) => {
...
+      data: {
+        id,
+        customName: url.customName,
+        originalUrl: url.originalUrl,
+        shortUrl: url.shortUrl,
+        createdAt: url.createdAt
+      }
+    });
+  } catch (error) {
+    logger.error(`Error occured while deleting url: ${error}`);
```
Comment (greatertomi) [inline src/controllers/url.ts]:
```suggestion
    logger.error(`Error occurred while deleting url: ${error}`);
```

**Review by greatertomi — CHANGES_REQUESTED** [review]
Clean code. 😍 . Fix those minor issues and you're good to go.

I'm thinking, it would be neat if we can centralize the error handling instead of having try-catch on every controller. This is optional, but we could work on it together if you want.

**Comment by PraiseTy** [comment]
> Clean code. 😍 . Fix those minor issues and you're good to go.
> 
> I'm thinking, it would be neat if we can centralize the error handling instead of having try-catch on every controller. This is optional, but we could work on it together if you want.

I don't mind working on this, I could book a meeting for tomorrow?

File: src/controllers/url.ts (hunk 1 above)
Comment (PraiseTy) [reply src/controllers/url.ts]: Alright, doing this now

**Comment by greatertomi** [comment]
> I don't mind working on this, I could book a meeting for tomorrow?

That was a fast response. You can fix a meeting for tomorrow.

**Comment by PraiseTy** [comment]
> > I don't mind working on this, I could book a meeting for tomorrow?
> 
> That was a fast response. You can fix a meeting for tomorrow.

Cool. I will. Also pushed the changes you requested

**Review by greatertomi — APPROVED** [review]

**Pull request merged**